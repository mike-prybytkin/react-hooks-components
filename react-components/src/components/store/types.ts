import { ReactNode } from 'react';
import { IProductCard } from 'share/types';

export interface StoreProviderProps {
  children: ReactNode;
}

export interface IStoreProviderContext {
  onSearch: (querySearch: string) => void;
  updateQuery: (querySearch: number) => void;
  querySearch: string;
  setQuerySearch: React.Dispatch<React.SetStateAction<string>>;
  queryPage: number;
  setQueryPage: React.Dispatch<React.SetStateAction<number>>;
  allPages: number;
  setAllPages: React.Dispatch<React.SetStateAction<number>>;
  cardsLimit: number;
  data: IProductCard[];
  setData: React.Dispatch<React.SetStateAction<IProductCard[]>>;
  isLoading: boolean;
  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
}
