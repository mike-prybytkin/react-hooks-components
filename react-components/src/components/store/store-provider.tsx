import React, { createContext, useState, ReactNode } from 'react';
import { IStoreProviderContext } from './types';
import { IProductCard } from 'share/types';

const StoreProviderContext = createContext({} as IStoreProviderContext);

interface StoreProviderProps {
  children: ReactNode;
}

const StoreProvider = (props: StoreProviderProps) => {
  const { children } = props;
  const [data, setData] = useState<IProductCard[]>([]);
  const [querySearch, setQuerySearch] = useState('');
  const [queryPage, setQueryPage] = useState(1);
  const [allPages, setAllPages] = useState(0);
  const cardsLimit = 20;

  const updateQuery = (queryPage: number) => {
    setQueryPage(queryPage);
  };

  const onSearch = (querySearch: string) => {
    setQuerySearch(querySearch);
  };

  return (
    <StoreProviderContext.Provider
      value={{
        data,
        setData,
        querySearch,
        setQuerySearch,
        queryPage,
        setQueryPage,
        allPages,
        setAllPages,
        cardsLimit,
        onSearch,
        updateQuery,
      }}
    >
      {children}
    </StoreProviderContext.Provider>
  );
};

export { StoreProvider, StoreProviderContext };
