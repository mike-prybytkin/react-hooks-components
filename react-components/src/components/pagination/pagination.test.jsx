import React from 'react';
import { render } from '@testing-library/react';
import Pagination from './pagination';
import { StoreProviderContext } from '../store/store-provider';

const setUp = (page, allPages) => {
  const updateQuery = jest.fn();
  return render(
    <StoreProviderContext.Provider value={{ updateQuery, queryPage: page, allPages: allPages }}>
      <Pagination />
    </StoreProviderContext.Provider>
  );
};

describe('Pagination component', () => {
  it('pagination should not render if one page is rendered', () => {
    const { container } = setUp(1, 1);
    expect(container.firstChild).toBeNull();
  });

  it('pagination should render if two pages are rendered', () => {
    const { container } = setUp(1, 2);
    expect(container.getElementsByClassName('numerical-button').length).toBe(2);
  });

  it('the first button should be active if the first page is rendered', () => {
    const { container } = setUp(1, 2);
    expect(container.getElementsByClassName('numerical-button')[0]).toHaveClass('active');
  });

  it('the PREV button should be disabled if the FIRST button is active', () => {
    const { container } = setUp(1, 2);
    expect(container.getElementsByClassName('pagination-prev-button')[0]).toHaveProperty(
      'disabled',
      true
    );
  });

  it('create pagination snapshot with buttons', () => {
    const { asFragment } = setUp(1, 5);
    expect(asFragment()).toMatchSnapshot();
  });
});
