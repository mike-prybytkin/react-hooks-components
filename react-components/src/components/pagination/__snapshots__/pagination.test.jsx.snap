// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component create pagination snapshot with buttons 1`] = `
<DocumentFragment>
  <div
    class="pagination-wrapper"
  >
    <button
      class="pagination-prev-button"
      disabled=""
    >
      prev
    </button>
    <div
      class="numerical-buttons-wrapper"
    >
      <button
        class="numerical-button active"
      >
        1
      </button>
      <button
        class="numerical-button"
      >
        2
      </button>
      <button
        class="numerical-button"
      >
        3
      </button>
      <button
        class="numerical-button"
      >
        4
      </button>
      <button
        class="numerical-button"
      >
        5
      </button>
    </div>
    <button
      class="pagination-next-button"
    >
      next
    </button>
  </div>
</DocumentFragment>
`;
